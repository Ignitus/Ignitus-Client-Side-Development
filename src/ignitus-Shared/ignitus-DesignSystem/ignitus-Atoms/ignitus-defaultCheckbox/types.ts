export type CheckBoxType = {
  label: string;
  value: string;
};
