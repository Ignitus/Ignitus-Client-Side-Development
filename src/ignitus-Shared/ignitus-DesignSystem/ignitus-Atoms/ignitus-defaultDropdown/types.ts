export type DropDownType = {
  label: string;
  options: string[];
  display: string;
};

export type DropDownProps = {
  display: string;
};
