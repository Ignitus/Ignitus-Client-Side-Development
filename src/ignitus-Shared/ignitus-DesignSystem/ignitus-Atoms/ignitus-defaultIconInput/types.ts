import { AppIcon } from '../../../types/iconsTypes/iconEnums';

export type DefaultIconInputProperties = {
  type: string;
  name: AppIcon;
  placeholder: string;
  state: string;
  width?: string;
  handleClick?: ((value: string) => void) | ((value: string) => null);
  handleChange: (value: string) => void;
};

export type DefaultPasswordInputProperties = {
  placeholder: string;
  state: string;
  handleChange: (value: string) => void;
};
