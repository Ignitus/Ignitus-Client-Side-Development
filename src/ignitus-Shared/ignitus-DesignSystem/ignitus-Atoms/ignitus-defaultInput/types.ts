export type InputType = {
  placeholder: string;
  type: string;
  name: string;
  height?: string;
  width?: string;
  handleChange?(inputValue: string): Function;
};
