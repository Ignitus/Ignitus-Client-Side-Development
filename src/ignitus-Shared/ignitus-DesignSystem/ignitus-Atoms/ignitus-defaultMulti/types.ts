export type InputTypes = {
  placeholder: string;
  name: string;
  maxHeightOfInput?: number;
  initialHeight?: string;
};
