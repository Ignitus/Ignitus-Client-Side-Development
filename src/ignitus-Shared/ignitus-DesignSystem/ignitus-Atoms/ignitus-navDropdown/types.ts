export type Props = {
  Options: Option[];
  Name: string;
};

type Option = {
  title: string;
};

export type StyleProps = {
  marginLeft?: boolean;
  marginRight?: boolean;
};
