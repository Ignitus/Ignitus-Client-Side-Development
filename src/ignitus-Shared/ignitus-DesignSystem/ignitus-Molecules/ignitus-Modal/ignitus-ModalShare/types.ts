export type ModalShareProps = {
  title: string;
  Share: string;
  Link: string;
  LinkedinColor: string;
  TwitterColor: string;
  RedditColor: string;
  FacebookColor: string;
  White: string;
  GreyOne: string;
};
