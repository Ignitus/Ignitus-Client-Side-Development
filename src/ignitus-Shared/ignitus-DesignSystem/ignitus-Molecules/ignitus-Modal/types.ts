export type Props = {
  secondary?: boolean;
};

export type ModalProps = {
  primaryText: string;
  secondaryText: string;
  title: string;
  buttonText: string;
};
