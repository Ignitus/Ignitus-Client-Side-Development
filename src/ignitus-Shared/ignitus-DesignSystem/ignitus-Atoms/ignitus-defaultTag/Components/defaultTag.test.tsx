import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';

import { DefaultTag1, DefaultTag2 } from './index';
import { InputTypes } from '../types';
import { CrossIcon } from '../../../ignitus-Assets/ignitus-Icons/crossIcon';

const props: InputTypes = {
  label: 'Label',
  color: '#000',
  background: '#fff',
};

let wrapper;

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

const styleCheck = () => {
  const span = wrapper.find('span');
  expect(span).toHaveStyleRule('color', props.color);
  expect(span).toHaveStyleRule('background', props.background);
};

it('should rendered', () => {
  const tree = render(<DefaultTag1 {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultTag1 />', () => {
  beforeEach(() => {
    wrapper = mount(<DefaultTag1 {...props} />);
  });

  it('should render span tag with label', () => {
    expect(wrapper.find('span').text()).toBe(props.label);
  });

  it('should render span with style passed in prop', styleCheck);
});

describe('<DefaultTag2 />', () => {
  beforeEach(() => {
    wrapper = mount(<DefaultTag2 {...props} />);
  });

  it('should render', () => {
    wrapper = render(<DefaultTag2 {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render span tag with label & CrossIcon', () => {
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find(CrossIcon).length).toBe(1);
  });

  it('should render span with style passed in prop', styleCheck);
});
