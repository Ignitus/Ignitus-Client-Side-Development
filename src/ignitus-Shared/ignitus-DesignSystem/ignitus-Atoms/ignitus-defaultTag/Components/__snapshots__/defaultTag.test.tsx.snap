// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultTag2 /> should render 1`] = `
.emotion-1 {
  background: #fff;
  color: #000;
  text-align: center;
  border-radius: 2rem;
  line-height: 1rem;
  padding: 1rem 2rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.5rem;
  cursor: pointer;
}

.emotion-0 {
  width: 1.4rem;
  fill: #000;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.03rem;
}

<span
  class="emotion-1"
  color="#000"
>
  <svg
    aria-label="CrossIcon"
    class="emotion-0"
    color="#000"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      CrossIcon
    </title>
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M18.3 5.71c-.39-.39-1.02-.39-1.41 0L12 10.59 7.11 5.7c-.39-.39-1.02-.39-1.41 0-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
    />
  </svg>
  Label
</span>
`;

exports[`should rendered 1`] = `
.emotion-0 {
  background: #fff;
  color: #000;
  text-align: center;
  border-radius: 2rem;
  line-height: 1rem;
  padding: 1rem 2rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.5rem;
  cursor: pointer;
}

<span
  class="emotion-0"
  color="#000"
>
  Label
</span>
`;
